common/journal_entries/imperia_infamy_rebalance_journal_entries.txt

```diff
@@ -86,31 +77,69 @@ je_is_pariah = {
 		infamy >= infamy_threshold:notorious
 	}
 	immediate = {
+		if = {
+			limit = {
+				NOT = {
+					exists = global_var:current_pariah
+				}
+			}
+			set_global_variable = {
+				name = current_pariah
+				value = this
+			}
+		}
 	}
 	complete = {
 		infamy < infamy_threshold:pariah
 	}
 	on_complete = {
+		remove_modifier ?= modifier_international_pariah
+		remove_modifier ?= modifier_pariah_economic_sanctions
+		remove_modifier ?= modifier_pariah_international_isolation
+		# If theres still another pariah on completion, set the next pariah to be the highest infamy pariah available..
 		if = {
 			limit = {
-				has_modifier = modifier_international_pariah
+				any_country = {
+					has_journal_entry = je_is_pariah
+					this != root
+				}
 			}
-			remove_modifier = modifier_international_pariah
-		}
-		if = {
-			limit = {
-				has_modifier = modifier_pariah_economic_sanctions
+			ordered_country = {
+				limit = {
+					has_journal_entry = je_is_pariah
+					this != root
+				}
+				order_by = infamy
+				position = 0
+				set_global_variable = {
+					name = current_pariah
+					value = this
+				}
 			}
-			remove_modifier = modifier_pariah_economic_sanctions
 		}
-		if = {
+		# Else, just clean the variable
+		else_if = {
 			limit = {
-				has_modifier = modifier_pariah_international_isolation
+				global_var:current_pariah ?= this
 			}
-			remove_modifier = modifier_pariah_international_isolation
+			remove_global_variable ?= current_pariah
 		}
 	}
 	on_monthly_pulse = {
+		effect = {
```

Comment:
A non-existent nation will not have a monthly journal entry pulse.

Reply:
No, but this is for setting the global_var to be a different existing pariah